packages/strapi/lib/middlewares/parser/index.js

```diff
@@ -1,41 +1,28 @@
 'use strict';
 
 const body = require('koa-body');
-const merge = require('merge-descriptors');
 const qs = require('qs');
 
 /**
  * Body parser hook
  */
-
-const addParser = app => {
-  merge(app.request, {
-    /**
+const addQsParser = app => {
+  Object.defineProperty(app.request, 'query', {
+    configurable: false,
+    enumerable: true,
+    /*
      * Get parsed query-string.
-     *
-     * @return {Object}
-     * @api public
      */
-    get query() {
-      var str = this.querystring;
-      if (!str) return {};
-
-      var c = (this._querycache = this._querycache || {});
-      var query = c[str];
-      if (!query) {
-        c[str] = query = qs.parse(str);
-      }
-      return query;
+    get: () => {
```

Comment:
you can use `get() { }` and `set() {}` directly :)

Reply:
Ok, removed arrow functions and used them directly 
(however in this case I have to say I find the version with arrows more readable)